test/autorange.jl

```diff
@@ -0,0 +1,114 @@
+                    for (i1,v1) in zip(1:2,[si(t[3]),ei(t[3])])
+                        @test A[:,:,i1,i2] == [
+                            CartesianIndex{4}((si(t[1]),si(t[2]),v1,v2))  CartesianIndex{4}((si(t[1]),ei(t[2]),v1,v2));
+                            CartesianIndex{4}((ei(t[1]),si(t[2]),v1,v2))  CartesianIndex{4}((ei(t[1]),ei(t[2]),v1,v2));
+                        ]
+                    end
+                end
+            else
+                error("This code must be unreached")
+            end
+        end
+    end
+
+    @testset "0-element and 1-element edge cases" begin
+        for t in ((),(1,),(1,1),(1,1,1))
+            N = length(t)
+            cr = CartesianRange(t)
+            ci = @inferred(ImageTransformations.CornerIterator(cr))
+            @test ci.start === cr.start
+            @test ci.stop === cr.stop
+            @test @inferred(collect(ci)) == collect(cr)
+            @test @inferred(size(ci)) === t
+            @test @inferred(length(ci)) === 1
+            @test length(collect(ci)) === 1
+            @test typeof(collect(ci)) <: Array{CartesianIndex{N},N}
+            @test all(collect(ci) .== [CartesianIndex{N}(t)])
+        end
+    end
+
+    @testset "arrays with singleton dimensions" begin
+        cr = CartesianRange((1,3))
+        ci = @inferred(ImageTransformations.CornerIterator(cr))
+        @test ci.start === cr.start
+        @test ci.stop === cr.stop
+        @test @inferred(size(ci)) === (1,2)
+        @test @inferred(length(ci)) === 2
+        @test length(@inferred(collect(ci))) === 2
+        @test typeof(collect(ci)) <: Array{CartesianIndex{2},2}
+        @test collect(ci) == [
+            CartesianIndex{2}((1,1)) CartesianIndex{2}((1,3))
+        ]
+        # two singleton dimensions
+        cr = CartesianRange((1,3,1,3))
+        ci = @inferred(ImageTransformations.CornerIterator(cr))
+        @test ci.start === cr.start
+        @test ci.stop === cr.stop
+        @test @inferred(size(ci)) === (1,2,1,2)
+        @test @inferred(length(ci)) === 4
+        @test length(@inferred(collect(ci))) === 4
+        @test typeof(collect(ci)) <: Array{CartesianIndex{4},4}
+        @test size(collect(ci)) === (1,2,1,2)
+        @test collect(ci)[1,:,1,:] == [
+            CartesianIndex{4}((1,1,1,1)) CartesianIndex{4}((1,1,1,3));
+            CartesianIndex{4}((1,3,1,1)) CartesianIndex{4}((1,3,1,3))
+        ]
+    end
+end
+
+@testset "autorange" begin
+    h, w = 10, 20
+    d = sqrt(h^2 + w^2)
+    tst_img = zeros(h,w)
+    # We compare the result of autorange against manually computed
+    # values using basic trigonometry
+    for ϕ in (pi/8, pi/4, pi/2)
+        rot = LinearMap(RotMatrix(ϕ))
+        rnge = @inferred ImageTransformations.autorange(tst_img, rot)
+        @test rnge[1].start == -floor(sin(ϕ)*w)
+        @test rnge[1].stop  ==  floor(cos(ϕ)*h)
+        @test rnge[2].start == 1
+        @test rnge[2].stop  == ceil(cos(atan(h/w)-ϕ)*d)
```

Comment:
Can someone maybe double check these four tests? They work, but I am a little confused as of why, since the code of `autorange` applies `floor` and `ceil` differently.